functions.py

```diff
@@ -101,17 +101,25 @@ def esami(update: Update, context: CallbackContext):
     check_log(update, context, "esami")
     if 'esami' in context.user_data: context.user_data['esami'].clear() #ripulisce il dict dell'user relativo al comando /esami da eventuali dati presenti
     else: context.user_data['esami'] = {} #crea il dict che conterrà i dati del comando /esami all'interno della key ['esami'] di user data
-    reply = get_esami_text_InlineKeyboard(context)
-    context.bot.sendMessage(chat_id=update.message.chat_id, text=reply[0], reply_markup=reply[1])
+
+    user_id = update.message.from_user.id
+    chat_id = update.message.chat_id
+    if chat_id != user_id: # forza ad eseguire il comando in una chat privata, anche per evitare di inondare un gruppo con i risultati
+        context.bot.sendMessage(chat_id=chat_id, text="Questo comando è utilizzabile solo in privato")
+        context.bot.sendMessage(chat_id=user_id, text="Dal comando esami che hai eseguito in un gruppo")

```

Comment:
ma se non metti un return l'esecuzione continua, quindi in teoria manda anche gli esami, o sbaglio?

Reply:
Yep, lo manda, ma se vedi il parametro della funzione "sendMessage" chat_id, il comando esami, con il piccolo messaggio di contesto iniziale, viene inoltrato nella chat privata dell'utente, mentre l'avviso viene mandato sul gruppo.

Reply:
perfetto ;-)